libraryBuilds/gson/build.gradle

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2015 the authors of j2objc-common-libs-e2e-test
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        // This is the build output of the plugin itself.
+        classpath fileTree(dir: '../../j2objc-gradle/build/libs', include: ['*.jar'])
+    }
+}
+
+apply plugin: 'java'
+apply plugin: 'com.github.j2objccontrib.j2objcgradle'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    j2objcTranslation 'com.google.code.gson:gson:2.3.1:sources'
+}
+
+j2objcConfig {
+    // Multiple Java filenames can indeed be identical.
+    filenameCollisionCheck false
```

Comment:
1.  prefixes.properties does not affect the filename, just the class name.  Also, these libraries in our repo should *never* use prefixes.properties.  Unless all transitive forward and backward dependencies of a library use the same prefixes.properties files, you are not guaranteed they will be able work with each other.

2.  We aren't using no-package-directories, however the plugin checks for filename collisions regardless of no-package-directories.  That is a bug in the plugin, filed: https://github.com/j2objc-contrib/j2objc-gradle/issues/467  - so until that is fixed it should indeed be set to false - because files in distinct directory do not indeed collide.